### DISCUSSION TITLE
The end of harpy

### DISCUSSION THREAD

**Discussion by robertodoering** [robertodoering/harpy#884] [General]
Twitter has now updated their terms of service to disallow all third party clients. This means that it's only a matter of time until api access for harpy gets revoked and the apps stops working.

With that in mind there is no insensitive for me or anyone else to continue developing harpy.
I won't archive this repository. If you find any mayor bugs feel free to still create issues for it and I'll have a look as long as the app is still working. However, feature requests and ideas are no longer necessary.

Thanks so much to the community for all the help and support. It's sad that we won't be able to see what harpy could have been, but the road to get to where we are now has been nothing but amazing.
harpy started as a fun little side project for me in late 2018 and remained a side project that loved to work on.

Thanks again for supporting the development over the years ❤️
~ Roberto

~~**Update (Feb, 14th):**
Due to upcoming changes to Twitter's API, harpy will soon stop working. Thus I've removed both the free and paid version from the Play Store.
If the changes still allow for free private api keys to be used, I'll add a download to the pro version to the [releases](https://github.com/robertodoering/harpy/releases) here on github.~~

**Update (Apr, 8th):** harpy can no longer access Twitter's api and has stopped working. rip

### Can I still use harpy?
~~We haven't gotten our api access restricted yet, so feel free to continue using the app while it works.~~
~~If you have your own api key, you can always use that and it should continue to work for the foreseeable future.~~

~~**Update (Feb, 9th):** access to the Twitter api will be behind a paywall after Februrary 13th and this marks the date harpy will stop working.~~

~~**Update (Feb, 14th):** after multiple delays, changes to the api coming "in the near future" will cause harpy to fully stop working.~~

No, unless you pay $100 / month for paid access to Twitter's api and use your key in harpy. Don't pay for it, it's not worth it. 

### Will development for harpy ever continue?
If Twitter backtracks their decision to ban third party Twitter apps in the near future, yes.

### Will you make harpy work for any alternative social media platform?
harpy itself will never connect to a non Twitter service. If I create an app for a different platform it will be separate from harpy.

---

If you have any other questions, feel free to use this discussions to ask them, or this Twitter thread: https://twitter.com/harpy_app/status/1616354066279403523.

**Comment by scottiefalkon** [comment]
Please make us a beautiful Mastodon app.

**Comment by Patrick-web** [comment]
Thank you and all the contributors for your wonderful work

**Comment by lazyb5** [comment]
I'd still like the pro features, I was about to buy it. But then twitter started acting shady with 3rd party apps. So I put off the purchase. Hopefully I can continue to use via custom api. Are there any plans to make pro free or lower the price?

**Reply by robertodoering** [reply]
Yeah, I have reduced the price by 50%.
If you have your own key, you can also just build the pro version yourself for free if you like.

**Reply by lazyb5** [reply]
> Yeah, I have reduced the price by 50%. If you have your own key, you can also just build the pro version yourself for free if you like.

Purchased

**Reply by robertodoering** [reply]
Thank you!

**Reply by lazyb5** [reply]
> Thank you!

Thank you for making the app. Twitter did you (and many other devs) wrong. Hopefully they get enough backlash and backtrack on this stupid policy change.

**Reply by RichDG** [reply]
Thank you for making your app. I have just installed it and paid for the pro version (I realise it may not keep on working but the fact that Twitter sent an e-mail asking if it was me that logged in on "chrome for android" gives me hope it will survive for a while as it seems to think it is the browser and hopefully it won't be blocking browser's access as that would mean the twitter website couldn't be accessed"). Hopefully Twitter will realise in time what an awful decision they have made and reverse it (unlikely but got to keep hoping).

**Reply by robertodoering** [reply]
Thanks @RichDG!

**Comment by ghost** [comment]
Thank you for developing this wonderful app. I've bought and used many Twitter 3rd party apps but yours was different. Didn't feel clunky or bloated

**Comment by OshekharO** [comment]
Hopefully twitter take back this decision 😃.

**Comment by Adithyan007hack** [comment]
It was a great journey with harpy I am using this app on two of my devices 
At the start of 2022, I was on a hunt for an open-source Twitter client and I stumbled upon harpy since then harpy is my favorite Twitter client. Hoping twitter will take back this decision

**Comment by Goooler** [comment]
Thanks!

**Comment by monee121** [comment]
Thank you for all the work you've done! Keeping my fingers crossed that Harpy stays off the radar and keeps functionality. Elon is an idiot!!

**Comment by dnestico** [comment]
Welp, it looks like for everyone using a personal key to get around the 3rd party ban, Elon has decided to now paywall the API completely so don't think this will be possible anymore after February 9th. :( https://twitter.com/TwitterDev/status/1621026986784337922?s=20&t=BlSYgirQMe5OekSnUty8PQ

![Screenshot_20230202-022542.png](https://user-images.githubusercontent.com/3034555/216259103-e6035841-cd15-4f2e-9782-3fcf025f4a2e.png)

**Reply by robertodoering** [reply]
That also means harpy will stop working on februrary 9th. rip.

**Reply by dnestico** [reply]
"Just $100 a month" This moron is insane. Smfh.

<img width="794" alt="Screen Shot 2023-02-02 at 5 36 48 PM" src="https://user-images.githubusercontent.com/3034555/216465453-7e0435f8-f6c0-45fa-a7c4-d7bbaa777493.png">

**Comment by scottiefalkon** [comment]
I just wanna say I warned of this many many weeks ago. I asked about a
Mastodon app weeks ago. Building an app or source is income of a closed
source infrastructure is always a huge risk. Harpy would make a beautiful
Mastodon app that many would gladly pay for.

**Reply by dnestico** [reply]
Personally not on Matsodon but I would 100% pay for a reddit app or even a web wrapper for facebook.

**Comment by scottiefalkon** [comment]
I'm not a fan of web wrappers and Facebook could still cause issues there.
A Reddit app would be nice, but there are already lean Material You apps
for that. Mastodon is growing and no one owns the code or API. It will not
be suddenly taken over or shit down. It's a growing space and a beautiful
app would sell well in that space. Not to mention donations. The open
source community can be quite amazing.

**Reply by nyabinary** [reply]
Mastodon 100%, open source product which means there no risk of the api every closing.

**Comment by heyboots** [comment]
It's really disappointing twitter forced this because they can't make a good decision to save their life. I'll at least be using the app until my API key inevitably stops working, and at that point I'll probably just dump the website entirely.

**Reply by heyboots** [reply]
finally stopped working today rip

**Reply by dnestico** [reply]
I'm using my own API key too and its still working. Damn....

**Reply by robertodoering** [reply]
@dnestico do you also still have access to the v2 endpoints / can you view profiles in harpy?

**Reply by dnestico** [reply]
@robertodoering looks like I can yes, do I check the Twitter dev portal or something for that? 

![Screenshot_20230608-191527.png](https://github.com/robertodoering/harpy/assets/3034555/22b7e76d-983d-47db-bb3f-83d511aebc40)

**Reply by robertodoering** [reply]
Yeah that means you do, which is weird because free access doesn't allow access to the endpoints used in the user profile (based on https://developer.twitter.com/en/portal/products/free).
For me everything but the profiles work with a custom key for now, since they still use the old v1 endpoints.

**Reply by dnestico** [reply]
Still up for me... for now I guess. weird.

It says V2 on the sidebar as you can see.
<img width="1053" alt="Screenshot 2023-06-08 at 7 33 52 PM" src="https://github.com/robertodoering/harpy/assets/3034555/692a35d8-5c52-4b1a-90c7-df4bf2856eed">

**Reply by dnestico** [reply]
@robertodoering This might finally be it for me... Opened the app and was logged out... Scared to try logging in tbh ![Screenshot_20230609-105727.png](https://github.com/robertodoering/harpy/assets/3034555/38f21322-1168-4385-b4e1-c926c3958c26)

**Reply by dnestico** [reply]
@robertodoering Update: I tried logging in and got redirected back to the login screen... Welp. I think it's done. :(

**Reply by robertodoering** [reply]
Same, can't use a private key anymore to login :/ I guess that's it then

**Comment by dnestico** [comment]
Looks like Harpy survives for another week... 

![Screenshot_20230208-192559.png](https://user-images.githubusercontent.com/3034555/217682771-c0062738-b324-43c6-a139-6fd7228cb38e.png)

**Comment by GoldrySeon** [comment]
Well, It's February 14, but Harpy is still working. And as far as I understand, Twitter still provides a free api, but with monthly restrictions on posting and authorization.

**Reply by robertodoering** [reply]
https://twitter.com/TwitterDev/status/1625234161010343941

They've delayed their api shenanigans by a "couple of days" again. So at least harpy will still function until then.

As for their free access that they'll offer, they mentioned it will only support a single authenticated user. Meaning if you have your own key you might be able to use it to use harpy but it can't ship with its own key.
https://twitter.com/TwitterDev/status/1623467618400374784

Seeing as how indecisive they are and the limited information that we have, we'll just have to wait.

**Comment by hlebushekhub** [comment]
Is it dead now?

**Reply by dnestico** [reply]
Still working for me with my own key, expecting that to die soon though with the death of the free APIs soon...

**Reply by ghost** [reply]
> Is it dead now? 

Not if you have like $40,000

**Reply by monee121** [reply]
Man I had just tweeted yesterday I was glad it was still working 😔

**Reply by dnestico** [reply]
Harpy will officially be dead soon, even if you have your own API key welp.

 ![Screenshot_20230407-194001.png](https://user-images.githubusercontent.com/3034555/230692560-14481b69-e342-4111-83a4-24a5ecc073c8.png)

**Reply by Still34** [reply]
Wouldn't adopting something like the official keys work as a workaround? https://gist.github.com/pvieito/ee6d2c8934a8f84b9aeb467585277b8a

**Comment by LuminarySage** [comment]
Please consider making a mastodon client. Your twitter client was the best one and I really wish for the same polish on a 3rd party mastodon client

**Comment by scottiefalkon** [comment]
Totally agree. You have a fantastic start on what would be the most
beautiful Mastodon app by far. Don't let Harpy go to waste.

**Comment by red-rouge22** [comment]
So sorry to see the end of Harpy Pro! It was THE twitter app. Feels like we're in a third world country, when choice and freedom are taken away! The official Twitter app in the G.Play store sucks the bag!!!! Nothing left accept the website👎
Roberto I hope you'll consider making a beautiful Mastodon app.  I'd happy support that project. Youre already half 
way there with Harpy.  My heart is broken re Harpy. Musk just destroys everything he touches. All the best to you!

**Comment by red-rouge22** [comment]
Hey!

How to create my own Api Key?

FWI: There are MANY new Twitter 3rd party apps in the Play Store. But not Harpy.
Why not? I mean if they can do it, so can you. Some work, and some dont.

Thanks for notifying me!!

red_rouge22

Sent from Proton Mail mobile

-------- Original Message --------

**Reply by robertodoering** [reply]
You can find a guide here: https://github.com/robertodoering/harpy/wiki/Twitter-api-key-setup

> FWI: There are MANY new Twitter 3rd party apps in the Play Store. But not Harpy.
Why not? I mean if they can do it, so can you. Some work, and some dont.

Because:
- It's against Twitter's TOS and they will revoke api access to any app that gets popular, like they already did with harpy and many others.
- Any third party client alive at the moment relies on free access to the legacy api that will most likely stop existing soon. Even paid access won't give you the access that is needed to build a proper app for multiple users.
- Twitter is trash

Read literally any news related to the shutdown of third party twitter clients to find out more

**Comment by red-rouge22** [comment]
Hang on.......Musk is ALWAYS changing mind and flip-flopping all over the place!

Other apps are still in the Store. I searched "Harpy Pro", and all these new apps came up!! Harpy of course did not, but Google did update my version.

I wont uninstall for the next 12 months.

Best...

Sent from Proton Mail mobile

-------- Original Message --------

**Comment by red-rouge22** [comment]
Thank you!

I understand now.

But being as how Musk changes his mind all the time and flip-flops all over the place, I'll hold onto the app for the next 12 months.
One never knows with him!

Best_

Sent from Proton Mail mobile

-------- Original Message --------

**Comment by ghost** [comment]
The API is working again! Don't know why though.

**Reply by monee121** [reply]
Yep I was just able to login....wonder how long it will last.  Thanks for letting us know!

**Reply by ghost** [reply]
The developer of FocusTwitter forked this and their version works fine, you can try it out [here](https://play.google.com/store/apps/details?id=allen.town.focus.twitter.x). [Source Code](https://github.com/allentown521/FocusTwitterX)

Yes, there are ads in the free version 😔

**Reply by dnestico** [reply]
Wait so do I need to download the forked version or can I just download the pro version I already have again from the playstore and log back in?

**Comment by red-rouge22** [comment]
Thanks, just tried it myself and its WORKING perfectly!!!! So happy!!!!
Hope it stays like this. Pass the word along.

Sent from Proton Mail mobile

-------- Original Message --------

**Comment by red-rouge22** [comment]
Does anyone know the url of the Google translator used in Harpy. Need to allow it.
Thanks

Sent from Proton Mail mobile

-------- Original Message --------

**Reply by robertodoering** [reply]
`translate.google.com`

**Comment by ghost** [comment]
They both work, just one is being maintained

**Reply by dnestico** [reply]
I see. I'm sure if the 3rd party app ban is officially lifted the dev for this would come back and continue maintaining it.

**Reply by dnestico** [reply]
Edit: I downloaded it, and hell yeah it's working. Finally twitter is back!... kinda

**Comment by ghost** [comment]
@robertodoering do you know why the API is working again?Even with that profile viewing is broken.

**Reply by robertodoering** [reply]
No idea
Officially "Only Media Upload and Login With Twitter" should be working:
https://developer.twitter.com/en/docs/twitter-api/getting-started/about-twitter-api

I'm guessing they shut down the legacy api temporarily, realized it broke to much or got pressure to boot it back up for the time being

**Comment by red-rouge22** [comment]
I searched the playstore for Harpy, because I had an old version, and did not find it, so downloaded FocusTwitter that was in the Harpy search results. However, Google updated my old version of Harpy to Harpy Pro. Signed in today!!
I wrote to Robert, about the many new Twitter apps appearing in the playstore, but he said they (Focus Twitter) would be discontinued, but hasnt happened over the 2 weeks Ive been using it, so far.

Sent from Proton Mail mobile

-------- Original Message --------

**Comment by hlebushekhub** [comment]
Oof it stopped working again

**Comment by ghost** [comment]
Same, harpy won't authenticate

**Comment by red-rouge22** [comment]
Same here too! But FocusTwitter is sill working. Was using both daily. I dont get that!!

Sent from Proton Mail mobile

-------- Original Message --------

**Comment by ghost** [comment]
This morning I was able to login to harpy again but everything is broken. I hope we find a fix to this soon! 
![Screenshot_2023-10-28-10-56-10-891_com robertodoering harpy free](https://github.com/robertodoering/harpy/assets/123662124/a1bd3d04-f484-41b5-9ea6-a00f8125c525)

**Comment by red-rouge22** [comment]
Hi,

I do too! Especially since FocusTwitter, 3rd party app is thriving beautifully, and even improving its features. How in the eorld can they do that, but you are not allowed to???

I know Im probably not understanding the technicalities here, but I use that app, always comparing it to Harpy. I still have it installed.

Sent from Proton Mail mobile

-------- Original Message --------

**Comment by yoshimo** [comment]
Now that Twitter announced a subscription without any ads, would that mean you get access to the API again? 
Without ad revenue requirements there is no need to force the official client anymore.

**Comment by red-rouge22** [comment]
Excellent!!!!

Thanks for the info!

Does Roberto apply again or what? I hope he will or the guys at Github that are interested.

Sent from Proton Mail mobile

-------- Original Message --------

**Comment by scottiefalkon** [comment]
I don't see how this is going to make any difference to third party clients
at all. Each user would have to pay for the ad free premium tier even if it
did allow outside access to that API. And I doubt it will.

**Reply by yoshimo** [reply]
it still is a lot cheaper than a single developer paying for every one of its users.

**Comment by scottiefalkon** [comment]
And what's the incentive for the developer to continue working on the app?
And who's to say even if this were to work that access wouldn't be turned
off again by Twitter without notice. Twitter is losing users monthly and
it's management is unreliable and it's future very much in doubt. I can see
no reason any developer would invest any time and money into apps for the
platform. It's over folks.
